Enable caching

athinakantis opened the pull request:
### **Summary**
To help performance and reduce load on the database I've enabled caching in the backend

EmilKronholm commented on UX/backend/src/services/pages.service.ts:
```diff
@@ -1,14 +1,26 @@
     try {
       const PAGES = ['hero', 'product', 'support'];
+      const CACHED_PAGES = await this.cacheManager.get('pages');
+      console.log('Cached pages: ', CACHED_PAGES)
```
Is there a risk this `console.log` will spam the console?

athinakantis replied:
Ah, silly me 😄 Totally missed that. Thanks for pointing it out!

EmilKronholm reviewed (approved):
Looks good to me! I havn't tested anything on my end, but I read through the logic and it seems valid. I just wanted to point out two concerns that I had:
____
- If a user post a review, it could take up to 1 hour to show up on the website? To avoid this you could always fetch a `lastUpdatedAt` and if not the same as the cached one we'd always fetch database for fresh data. Is this okay?

- Cache logic could make the code messy and would be better off abstracted somewhere, but I don't know if this will really be a problem in the product website (as I don't know frequently cache will be used) :)
____
Thanks Athina for your hard work!

athinakantis commented:
> Looks good to me! I havn't tested anything on my end, but I read through the logic and it seems valid. I just wanted to point out two concerns that I had:
> 
> * If a user post a review, it could take up to 1 hour to show up on the website? To avoid this you could always fetch a `lastUpdatedAt` and if not the same as the cached one we'd always fetch database for fresh data. Is this okay?
> * Cache logic could make the code messy and would be better off abstracted somewhere, but I don't know if this will really be a problem in the product website (as I don't know frequently cache will be used) :)
> 
> Thanks Athina for your hard work!

Thanks Emil :)
I will have a look into checking when the table was last updated, it sounds very logical. I'm looking into it now 👀 Great feedback as always, appreciate your attention to detail!

athinakantis commented:
### I made a few changes:

- Created a `isCacheFresh(LAST_UPDATED, MAX_AGE)` function in `/utils`
- Added a function to get data from DB of when table was last updated
- Updated `tsconfig.build.json` to include the new `/utils` folder

EmilKronholm commented:
Everything looks good! Feel free to merge if you are ready! :) 🥇 💯
